ImperatorToCk2/Directories.java

```diff
@@ -0,0 +1,130 @@
+     int aqv = 0;
+     boolean  endOrNot2 = true;
+     String VM = "\\";
+     VM = VM.substring(0);
+     String mainModFolder = Dir+VM+modName;
+     File f = new File(mainModFolder);
+        f.mkdir();
+        
+     File f2 = new File(mainModFolder+VM+"common");
+        f2.mkdir();
+     
+     File f3 = new File(mainModFolder+VM+"events");
+        f3.mkdir();   
+        
+     File f4 = new File(mainModFolder+VM+"gfx");
+        f4.mkdir();   
+        
+     File f5 = new File(mainModFolder+VM+"history");
+        f5.mkdir();   
+        
+     File f6 = new File(mainModFolder+VM+"localisation");
+        f6.mkdir(); 
+    // Sub folders
+    
+     File f7 = new File(mainModFolder+VM+"history"+VM+"characters");
+        f7.mkdir();  
+        
+     File f8 = new File(mainModFolder+VM+"history"+VM+"provinces");
+        f8.mkdir();  
+        
+     File f9 = new File(mainModFolder+VM+"history"+VM+"technology");
+        f9.mkdir();  
+      
+     File f10 = new File(mainModFolder+VM+"history"+VM+"titles");
+        f10.mkdir();  
+        
+     File f11 = new File(mainModFolder+VM+"history"+VM+"titles");
+        f11.mkdir();  
+        
+     File f12 = new File(mainModFolder+VM+"common"+VM+"cultures");
+        f12.mkdir();  
+        
+     File f13 = new File(mainModFolder+VM+"common"+VM+"dynasties");
+        f13.mkdir();  
+        
+     File f14 = new File(mainModFolder+VM+"common"+VM+"landed_titles");
+        f14.mkdir();  
+        
+     File f15 = new File(mainModFolder+VM+"common"+VM+"religions");
+        f15.mkdir(); 
+        
+     File f16 = new File(mainModFolder+VM+"common"+VM+"bookmarks");
+        f16.mkdir(); 
+        
+     File f17 = new File(mainModFolder+VM+"eu4_converter");
+        f17.mkdir();
+        
+     File f18 = new File(mainModFolder+VM+"gfx"+VM+"interface");
+        f18.mkdir(); 
+        
+     File f19 = new File(mainModFolder+VM+"eu4_converter"+VM+"copy");
+        f19.mkdir();
+        
+     File f20 = new File(mainModFolder+VM+"eu4_converter"+VM+"copy"+VM+"common");
+        f20.mkdir();
+        
+     File f21 = new File(mainModFolder+VM+"eu4_converter"+VM+"copy"+VM+"common"+VM+"religions");
+        f21.mkdir();
+        
+     File f22 = new File(mainModFolder+VM+"eu4_converter"+VM+"copy"+VM+"common"+VM+"cultures");
+        f22.mkdir();
+        
+     File f23 = new File(mainModFolder+VM+"common"+VM+"bloodlines");
+        f23.mkdir();  
+        
+    return aqv;
+    }
+    
+    public static String descriptors(String Dir, String modName, String Dir2) throws IOException 
+    {
+        //Each mod requires 2 .mod "descriptor" files so the game launcher can
+        //read the mod files as a mod
+        String VM = "\\";
+        char VMq = '"';
+        String VN = "//";
+       VN = VN.substring(0);
+        VM = VM.substring(0);
+        int aqq = 1;
+        String mainModFolder = Dir+VM+modName;
+        String dotModFolder = Dir2+VN+modName;
```

Comment:
No need for Dir2, you can instead do 
```
String descriptorFile = Dir + "/" + modname + "/descriptor.mod";
```

Reply:
Also, looks like you aren't currently outputting descriptor.mod. You should do so.